api/bucketsd/service.go

```diff
@@ -2391,116 +2391,125 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
-		// that is the default value, get the default address from the FFS instance
-		storageConfig.Cold.Filecoin.Addr = defConf.Cold.Filecoin.Addr
+		// that is the default value, get the default address from the storage profile
+		storageConfig.Cold.Filecoin.Address = defConfRes.DefaultStorageConfig.Cold.Filecoin.Address
 	}
 
-	// Check that FFS wallet addr balance is > 0, if not, fail fast.
-	bal, err := s.PowergateClient.Wallet.Balance(ctx, storageConfig.Cold.Filecoin.Addr)
+	// Check that storage profile wallet addr balance is > 0, if not, fail fast.
+	balRes, err := s.PowergateClient.Wallet.Balance(ctx, storageConfig.Cold.Filecoin.Address)
 	if err != nil {
 		return nil, fmt.Errorf("getting powergate wallet address balance: %s", err)
 	}
-	if bal == 0 {
+	bal, ok := new(big.Int).SetString(balRes.Balance, 10)
+	if !ok {
+		return nil, fmt.Errorf("error converting balance %v to big int", balRes.Balance)
+	}
+	if bal.Cmp(big.NewInt(0)) == 0 {
 		return nil, buckets.ErrZeroBalance
 	}
 
-	// Archive pushes the current root Cid to the corresponding FFS instance of the bucket.
+	// Archive pushes the current root Cid to the corresponding storage profile of the bucket.
 	// The behaviour changes depending on different cases, depending on a previous archive.
-	// 0. No previous archive or last one aborted: simply pushes the Cid to the FFS instance.
+	// 0. No previous archive or last one aborted: simply pushes the Cid to Powergate.
 	// 1. Last archive exists with the same Cid:
 	//   a. Last archive Successful: fails, there's nothing to do.
 	//   b. Last archive Executing/Queued: fails, that work already starting and is in progress.
 	//   c. Last archive Failed/Canceled: work to do, push again with override flag to try again.
-	// 2. Archiving on new Cid: work to do, it will always call Replace(,) in the FFS instance.
-	var jid ffs.JobID
+	// 2. Archiving on new Cid: work to do, it will always call Replace(,).
+	var jid string
 	firstTimeArchive := ba.Archives.Current.JobID == ""
 	if firstTimeArchive || ba.Archives.Current.Aborted { // Case 0.
 		// On the first archive, we simply push the Cid with
 		// the default CidConfig configured at bucket creation.
-		jid, err = s.PowergateClient.FFS.PushStorageConfig(
-			ctxFFS,
-			p.Cid(),
+		res, err := s.PowergateClient.StorageConfig.Apply(
+			ctxPow,
+			p.Cid().String(),
 			pow.WithStorageConfig(storageConfig),
 			pow.WithOverride(true),
 		)
 		if err != nil {
 			return nil, fmt.Errorf("pushing config: %s", err)
 		}
+		jid = res.JobId
 	} else {
 		oldCid, err := cid.Cast(ba.Archives.Current.Cid)
 		if err != nil {
 			return nil, fmt.Errorf("parsing old Cid archive: %s", err)
 		}
 
+		statusName, found := userPb.JobStatus_name[int32(ba.Archives.Current.JobStatus)]
+		if !found {
+			return nil, fmt.Errorf("invalid job status %v", ba.Archives.Current.JobStatus)
+		}
+
+		status := userPb.JobStatus(ba.Archives.Current.JobStatus)
```

Comment:
Turns out that the `ffs.JobStatus` raw values are the same as the proto enum `userPb.JobStatus` raw values. So that makes it easy to not use the `ffs` version. This is desirable because the Powergate client no longer returns `ffs` types from its API.